exploration/beauty-contest.md

```diff
@@ -121,6 +162,40 @@ Hello {$var}, you have a {$foo}
 {#match {$foo :function option=value}{$bar :function option=value}} {#when a b} {{  {$foo} is {$bar}  }} {#when x y} {{  {$foo} is {$bar}  }} {#when * *} {|  |}{$foo} is {$bar}{|  |}
 ```
 
+## 1b. Templating Language
+
+Same as 1, but directly inspired by existing templating languages.
+Leverages the familiarity with established templating solutions,
+such as [Jinja](https://jinja.palletsprojects.com/en/3.1.x/), [Twig](https://twig.symfony.com), and [Liquid](https://shopify.github.io/liquid/).
+
+```
+Hello world!
+
+Hello {{ $user }}
+
+{% input $var :function option=value %}
+Hello {{ $var }}
+
+{% input $var :function option=value %}
+{% local $foo = $bar :function option=value %}
+Hello {{ $var }}, you have a {{ $foo }}
+
+{% match {$foo} {$bar} %}
+{% when foo bar %} Hello {{ $foo }} you have a {{ $var }}
+{% when * * %} {{ $foo }} hello you have a {{ $var }}
+
+{% match {$foo :function option=value} {$bar :function option=value} %}
+{% when a b %} {|  {{$foo}} is {{$bar}}  |}
+{% when x y %} {|  {{$foo}} is {{$bar}}  |}
```

Comment:
These aren't right. They should be:

```suggestion
{% when x y % }}{||}  {{$foo}} is {{$bar}}  {||}
```

(I'm not sure how to type the `a b` example here, as you're using double-`{{` for variables. Quoting the whole pattern would have to be revisited as a topic. Conversely, I'm not sure we need to double the brackets for variables. I understand that you're being super-consistent with templating and so shouldn't read too much into it.

I'd probably add a comment about what I just said above to the description.

Reply:
No, this is intended (but we should discuss exact sigils). In this approach, all sigils are double characters: `{% %}` for statements, `{{ }}` for placeholders, and `{| |}` for quoted patterns. OTOH, an empty string literal in a placeholder would be `{{ || }}`.